File: CodingSamples/Java/Platform-Enterprise/ThreadTest/demo1/Program3.java
class Program {

    private static ThreadLocal<String> user = new ThreadLocal<>();

    private static void handleJob(int jid) {
        System.out.printf("Thread<%x> has accepted job<%d> for %s%n", Thread.currentThread().hashCode(), jid, user.get());
        Activity.perform(jid);
        System.out.printf("Thread<%x> has finished job<%d> for %s%n", Thread.currentThread().hashCode(), jid, user.get());
    }

    public static void main(String[] args) throws Exception {
        int n = args.length > 0 ? Integer.parseInt(args[0]) : 1;
        Thread child = new Thread(() -> {
            user.set("Jack");
            handleJob(n);
        });
        if(n > 5)
            child.setDaemon(true);
            //make child a background thread so that JVM doesn't
            //have to wait for it to exit
        child.start();
        user.set("Jill");
        handleJob(2);
    }
}

File: CodingSamples/DotNet/Platform-Enterprise/ThreadTest/DemoApp1/Program3.cs
namespace DemoApp;

class Program
{
    [ThreadStatic]
    static string user;

    static void HandleJob(int jid)
    {
        Console.WriteLine("Thread<{0}> has accepted job<{1}> for {2}", Environment.CurrentManagedThreadId, jid, user);
        Activity.Perform(jid);
        Console.WriteLine("Thread<{0}> has finished job<{1}> for {2}", Environment.CurrentManagedThreadId, jid, user);
    }

    static void Main(string[] args)
    {
        int n = args.Length > 0 ? int.Parse(args[0]) : 1;
        var child = new Thread(() => 
        {
            user = "Jack";
            HandleJob(n);
        });
        child.IsBackground = n > 5; //CLR does not wait for a background thread to exit
        child.Start();
        user = "Jill";
        HandleJob(2);
    }
}
